src/adapter/src/catalog/storage.rs

```diff
@@ -1381,7 +1419,7 @@ impl<'a> Transaction<'a> {
     }
 
     pub fn remove_role(&mut self, name: &str) -> Result<(), Error> {
-        let n = self.roles.delete(|_k, v| v.name == name).len();
+        let n = self.roles.delete(|_k, v| v.role.name == name).len();
```

Comment:
Should this function also check that the role key is of type User? Do system roles ever get removed? (If so, this probably doesn't need to be in this PR, I'm just worried about it now.)